### PR TITLE
allow size to be generic Sequence

### PR DESCRIPTION
**Pull request by wangg12**
For example, when we pass a `size` arg in a config created by omegaconf, only allowing `list, tuple` will raise an error.

### CODE REVIEW

File: torchvision/transforms/v2/_geometry.py
Comment (NicolasHug) [inline torchvision/transforms/v2/_geometry.py:138]:
```suggestion
        elif isinstance(size, Sequence) and len(size) in {1, 2}:
```

**Review by NicolasHug — APPROVED** [review]
Thanks @wangg12 , LGTM (I'll self-commit my suggestion below and wait for the CI before merging)

Technically we should probably be using `Sequence` from `collections.abc` instead of from `typing`, but maybe this is better addressed in a more general PR. @pmeier any opinion?

**Review by pmeier — COMMENTED** [review]
> Technically we should probably be using `Sequence` from `collections.abc` instead of from `typing`, but maybe this is better addressed in a more general PR.

`typing.Sequence` is actually an [alias](https://github.com/python/cpython/blob/edf701c4d83f399bff8612c34fef95da7663945e/Lib/typing.py#L1465) for `collections.abc.Sequence`. Meaning, we can do it for clarity, but there is no functional difference.

---

Should we also fix the error message below to reflect that this supports arbitrary sequences now?

https://github.com/pytorch/vision/blob/1a9ff0d728787df66a372283328158879805dcad/torchvision/transforms/v2/_geometry.py#L140-L143

**Comment by NicolasHug** [comment]
Linux unittests and linters are happy, merging. Thanks @wangg12 !

**Pull request merged**
